app/src/main/java/com/zulip/android/activities/ZulipActivity.java

```diff
@@ -1302,12 +1309,32 @@ private Cursor makeStreamCursor(CharSequence streamName)
                 .getDao(Stream.class)
                 .queryRaw(
                         "SELECT rowid _id, * FROM streams WHERE "
+                                + Stream.NAME_FIELD
+                                + " LIKE ? ESCAPE '\\' ORDER BY "
+                                + Stream.NAME_FIELD + " COLLATE NOCASE",
+                        DatabaseHelper.likeEscape(streamName.toString()) + "%")
+                .closeableIterator().getRawResults()).getRawCursor();
+    }
+
+
+    private boolean isSubscribed(CharSequence streamName)
+            throws SQLException {
+        if (streamName == null) {
+            streamName = "";
+        }
+
+        Cursor cursor =  ((AndroidDatabaseResults) app
+                .getDao(Stream.class)
+                .queryRaw(
+                        "SELECT rowid _id, * FROM streams WHERE "
                                 + Stream.SUBSCRIBED_FIELD + " = 1 AND "
                                 + Stream.NAME_FIELD
                                 + " LIKE ? ESCAPE '\\' ORDER BY "
                                 + Stream.NAME_FIELD + " COLLATE NOCASE",
                         DatabaseHelper.likeEscape(streamName.toString()) + "%")
                 .closeableIterator().getRawResults()).getRawCursor();
+
+        return cursor.getCount()==1;
```

Comment:
just to be safe (as you have used `like` in where clause) use `cursor.getCount() != 0` 
This change won't be needed if you use `where Stream.NAME_FIELD =`.